src/main/kotlin/com/terraformation/backend/admin/AdminModulesController.kt

```diff
@@ -24,23 +42,154 @@ import org.springframework.web.servlet.mvc.support.RedirectAttributes
+      @PathVariable moduleId: String,
+      redirectAttributes: RedirectAttributes
+  ): String {
+    val module =
+        try {
+          moduleStore.fetchOneById(ModuleId(moduleId))
+        } catch (e: ModuleNotFoundException) {
+          log.warn("Module not found")
+          redirectAttributes.failureMessage = "Import failed: ${e.message}"
+          return redirectToModulesHome()
+        }
+
+    val moduleProjects = module.cohorts.flatMap { it.projects!!.toList() }
+
+    val cohortNames =
+        module.cohorts.associate { it.cohortId to cohortStore.fetchOneById(it.cohortId).name }
+
+    val projectNames = moduleProjects.associateWith { projectStore.fetchOneById(it).name }
+
+    // cohort name - project name
+    val cohortProjectNames =
+        moduleProjects.associateWith {
+          val project = projectStore.fetchOneById(it)
+          cohortStore
+              .fetchOneById(participantStore.fetchOneById(project.participantId!!).cohortId!!)
+              .name + " - " + project.name
+        }
+
+    model.addAttribute("canManageModules", currentUser().canManageModules())
+    model.addAttribute("cohortNames", cohortNames)
+    model.addAttribute("cohortProjectNames", cohortProjectNames)
+    model.addAttribute("dateFormat", dateFormat)
+    model.addAttribute("module", module)
+    model.addAttribute("moduleProjects", moduleProjects)
+    model.addAttribute("projectNames", projectNames)
+    model.addAttribute("workshopInfo", module.eventDescriptions[EventType.Workshop])
+    model.addAttribute("oneOnOneInfo", module.eventDescriptions[EventType.OneOnOneSession])
+    model.addAttribute("liveSessionInfo", module.eventDescriptions[EventType.LiveSession])
+    model.addAttribute(
+        "workshopEvents", module.eventSessions[EventType.Workshop] ?: emptyList<EventModel>())
+    model.addAttribute(
+        "oneOnOneEvents",
+        module.eventSessions[EventType.OneOnOneSession] ?: emptyList<EventModel>())
+    model.addAttribute(
+        "liveSessionEvents", module.eventSessions[EventType.LiveSession] ?: emptyList<EventModel>())
+
+    return "/admin/moduleView"
+  }
+
+  @PostMapping("/modules/{moduleId}/createEvent")
+  fun createModuleEvent(
+      model: Model,
+      @PathVariable moduleId: String,
+      @RequestParam startTime: String,
+      @RequestParam endTime: String,
+      @RequestParam eventTypeId: Int,
+      @RequestParam meetingUrl: URI?,
+      @RequestParam recordingUrl: URI?,
+      @RequestParam slidesUrl: URI?,
+      @RequestParam toAdd: List<ProjectId>?,
+      redirectAttributes: RedirectAttributes
+  ): String {
+    eventStore.create(
+        ModuleId(moduleId),
+        EventType.forId(eventTypeId) ?: throw IllegalArgumentException("Event Type not recognized"),
+        dateStringToInstant(startTime),
+        dateStringToInstant(endTime),
+        meetingUrl,
+        recordingUrl,
+        slidesUrl,
+        toAdd?.toSet() ?: emptySet())
+    return redirectToModule(ModuleId(moduleId))
```

Comment:
Setting the success message here to acknowledge the creation (and maybe to include the event ID) would be good so the user knows it succeeded.

It'd also be good to wrap the create call in a try block and set a failure message if it throws an exception for whatever reason.

Same with the update and delete endpoints.